pilot/cmd/istioctl/controlplane/cmd.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// Command returns the "control-plane" subcommand for istioctl.
+func Command(istioNamespaceFlag *string) *cobra.Command {
+	var (
+		features          *[]string
+		out               string
+		helmChartLocation string
+	)
+
+	install := defaultInstall()
+	cmd := &cobra.Command{
+		Use:   "control-plane",
```

Comment:
'control-plane' is not so clear. Maybe call it 'gen-yamls'? Strictly speaking, ingress is not control plane. The name of the command should be consistent with what it is doing.

Reply:
What about `gen-deploy` or `gen-deployment`?

Reply:
gen-yamls (or better gen-kube-yamls) is less confusing. Deployment could be confused with K8S deployment object.

Reply:
The primary out _is_ kubernetes deployment objects (along with the stuff required for RBAC). `gen-yamls` is not descriptive: what YAML are you generating? What about when we extend the tool to produce scripts for VM based deployments?